typescript/lib/bulk_command_execution/bulk_command_serializer.ts

```diff
@@ -0,0 +1,42 @@
+import { V0BulkCommands } from "./v0_bulk_command_api/v0_bulk_commands";
+import { SchemaVersion, V0 } from "./bulk_command_schema_version";
+import { ok, err, Result } from "neverthrow";
+
+const latestSchemaVersion = V0;
+
+class VersionedBulkCommandsDocument {
+	private readonly schemaVersion: SchemaVersion;
+
+    constructor(schemaVersion: SchemaVersion) {
+        this.schemaVersion = schemaVersion;
+    }
+}
+
+class SerializableBulkCommandsDocument extends VersionedBulkCommandsDocument {
+	private readonly body: V0BulkCommands; //TODO - I should generalize this to any inteface? Need to look into this, or would need some kind of parent interface
```

Comment:
Yep, this should actually be `any` because when we have future versions, this `body` could be EITHER `V0BulkCommands` or `V1BulkCommands`